README.md

```diff
@@ -0,0 +1,71 @@
+Generate code from CDDL description
+===================================
+
+CDDL is a human-readable description language defined in an IETF RFC (https://tools.ietf.org/html/rfc8610).
+Using the CMake function `cddl_source()`, you can autogenerate C code that validates and decodes CBOR encoded instances of the structure described in a given .cddl file.
+
+The code generation is performed by the Python script cddl_gen.py in scripts/.
+The generated code depends on a simple CBOR decoding library (cbor_decode) in src/.
+There are tests for the code generation in tests/.
+The tests require Zephyr (If your shell is set up to build Zephyr samples, the tests should also build)
+
+Features
+========
+
+The generated code consists of:
+ - A header file containing typedefs corresponding to the types described in the CDDL, as well as functions declarations for decoding functions requested to be exposed (ENTRY_TYPES).
+ - A C file containing all the decoding code.
+   The decoding code is divided into functions (internal unless the type is an ENTRY_TYPE), and each function generally contains a single massive `if` statement which strings together calls into the cbor_decode library.
+
+CDDL allows specifying structures both loosely and strictly, both in terms of content (e.g. values of ints or strings), and structure (e.g. the type of an entry, the number of members in a list).
+The generated code will validate the input, which means that it will check all the restriction set in the CDDL description, and fail if a restriction is broken.
+This is especially important for security applications, but also useful to avoid segfaults and other errors as a result of malformed input.
+
+The cbor_decode library does most of the actual extraction and validation of individual values.
+
+The `cddl_source()` function sets up build steps necessary to call the script on the provided CDDL file, and adds the generated file as well as the cbor_decode library to your project.
```

Comment:
cddl_build_source() a better name?

Reply:
Maybe cddl_generate()? I'm not sold either way, any more opinions on this?

Reply:
@frkv please join the discussion here: https://github.com/oyvindronningstad/cddl_gen/pull/1#pullrequestreview-353560174